<!doctype html>
<html>
<head>
<meta charset="utf-8">
<title>Untitled Page</title>
<meta name="generator" content="WYSIWYG Web Builder 19 - https://www.wysiwygwebbuilder.com">
<link href="fontawesome6.min.css" rel="stylesheet">
<link href="Workshop.css" rel="stylesheet">
<link href="index.css" rel="stylesheet">
<script src="anime.min.js"></script>
<script src="polymorph.min.js"></script>
<script src="popper.min.js"></script>
<script src="bootstrap.min.js"></script>
<script src="wb.affix.min.js"></script>
<script src="scrollspy.min.js"></script>
<script src="jquery-3.7.1.min.js"></script>
<script>
document.addEventListener('DOMContentLoaded', function(event)
{
   var shapePath = document.querySelector('#MorphingIcon2-shape');
   var paths = ["M0 0 L370 0 L370 8056 L0 8056Z "];
   var interpolator = [];
   interpolator[0] = polymorph.interpolate([paths[0], paths[0]]);
   let progress = { prop: 0 };
   var pathIndex = 0;
   anime({
      targets: progress,
      prop: 1,
      delay: 0,
      duration: 2004,
      easing: 'linear',
      loop: true,
      direction: 'normal',
      update(anim) 
      {
         shapePath.setAttribute('d', interpolator[pathIndex](progress.prop));
         if (anim.progress == 100)
         {
            pathIndex++;
            if (pathIndex > 0)
               pathIndex = 0;
         }
      }
   });
});
</script>
<script>
document.addEventListener('DOMContentLoaded', function(event)
{
   var shapePath = document.querySelector('#MorphingIcon1-shape');
   var paths = ["M0 0 L370 0 L370 70 L0 70Z "];
   var interpolator = [];
   interpolator[0] = polymorph.interpolate([paths[0], paths[0]]);
   let progress = { prop: 0 };
   var pathIndex = 0;
   anime({
      targets: progress,
      prop: 1,
      delay: 0,
      duration: 2004,
      easing: 'linear',
      loop: true,
      direction: 'normal',
      update(anim) 
      {
         shapePath.setAttribute('d', interpolator[pathIndex](progress.prop));
         if (anim.progress == 100)
         {
            pathIndex++;
            if (pathIndex > 0)
               pathIndex = 0;
         }
      }
   });
});
</script>
<script src="wwb19.min.js"></script>
<script>
document.addEventListener('DOMContentLoaded', function(event)
{
   var DropdownMenu1_dropdownToggle = document.querySelectorAll('#DropdownMenu1 .dropdown-toggle');
   DropdownMenu1_dropdownToggle.forEach(item => 
   {
      var dropdown = new bootstrap.Dropdown(item, {popperConfig:{placement:item.getAttribute('data-bs-placement')}});
   });
   var DropdownMenu1_dropdown = document.querySelectorAll('#DropdownMenu1 .dropdown');
   DropdownMenu1_dropdown.forEach(item => 
   {
      item.addEventListener('shown.bs.dropdown', function(e)
      {
         e.currentTarget.classList.add('show');
      });
      item.addEventListener('hidden.bs.dropdown', function(e)
      {
         e.currentTarget.classList.remove('show');
      });
   });
   document.addEventListener('click', function (e)
   {
      var isChildOfDropdownMenu = false;
      var target = e.target;
      while (target !== null)
      {
         if (target.classList && target.classList.contains('DropdownMenu1-navbar-collapse') && target.classList.contains('show'))
         {
            isChildOfDropdownMenu = true;
            break;
         }
         target = target.parentNode;
      }
      if (isChildOfDropdownMenu)
      {
         if (e.target.tagName.toLowerCase() === 'a' && !e.target.classList.contains('dropdown-toggle'))
         {
            const dropdownMenu = document.querySelector('.DropdownMenu1-navbar-collapse');
            if (dropdownMenu)
            {
               const collapseInstance = bootstrap.Collapse.getInstance(dropdownMenu);
               if (collapseInstance)
               {
                  collapseInstance.hide();
               }
            }
         }
      }
   });
});
</script>
</head>
<body data-bs-spy="scroll">
<div id="wb_MorphingIcon2">
<svg id="MorphingIcon2" height="100%" width="100%" style="position:absolute;height:100%;width:100%;left:0;top:0;overflow:visible;" viewBox="0 0 370 8056" preserveAspectRatio="none" version="1.1" xmlns="http://www.w3.org/2000/svg">
<defs>
   <path id="MorphingIcon2-shape" d="M0 0 L370 0 L370 8056 L0 8056Z " vector-effect="non-scaling-stroke" />
</defs>
<use xlink:href="#MorphingIcon2-shape" data-href="#MorphingIcon2-shape" style="fill:#606060;stroke:#0000FF;stroke-width:1"/>
</svg>
</div>
<div id="wb_MorphingIcon1" style="position:absolute;left:0px;top:0px;width:370px;height:70px;z-index:1;">
<svg id="MorphingIcon1" height="100%" width="100%" style="position:absolute;height:100%;width:100%;left:0;top:0;overflow:visible;" viewBox="0 0 370 70" preserveAspectRatio="none" version="1.1" xmlns="http://www.w3.org/2000/svg">
<defs>
   <path id="MorphingIcon1-shape" d="M0 0 L370 0 L370 70 L0 70Z " vector-effect="non-scaling-stroke" />
</defs>
<use xlink:href="#MorphingIcon1-shape" data-href="#MorphingIcon1-shape" style="fill:#0D6EFD;stroke:#0000FF;stroke-width:1"/>
</svg>
</div>
<label for="" id="Label1" onclick="ShowObject('wb_Text1', 1);return false;" style="position:absolute;left:42px;top:1px;width:318px;height:54px;line-height:54px;z-index:2;">Metagenomics Workshop</label>
<div id="wb_Software_installation" style="position:absolute;left:359px;top:980px;width:52px;height:51px;z-index:3;">
<div id="Software_installation" style="visibility:hidden;"></div>
</div>
<div id="wb_Text2" style="position:absolute;left:406px;top:1153px;width:859px;height:106px;text-align:justify;z-index:4;">
<span style="color:#000000;font-family:'Times New Roman';font-size:16px;"><br>What is Windows Subsystem for Linux (WSL)?<br><br>The Windows Subsystem for Linux (WSL) is a feature of the Windows operating system that enables you to run a Linux file system, along with Linux command-line tools and GUI apps, directly on Windows, alongside your traditional Windows desktop and apps.<br></span></div>
<div id="wb_Install_WSL" style="position:absolute;left:367px;top:1116px;width:42px;height:29px;z-index:5;">
<div id="Install_WSL" style="visibility:hidden;"></div>
</div>
<div id="wb_Text3" style="position:absolute;left:409px;top:1267px;width:707px;height:32px;z-index:6;">
<span style="color:#000000;font-family:'Times New Roman';font-size:29px;"><strong>Prerequisites</strong></span></div>
<div id="wb_Text4" style="position:absolute;left:406px;top:1320px;width:859px;height:18px;text-align:justify;z-index:7;">
<span style="color:#000000;font-family:'Times New Roman';font-size:16px;">You must be running Windows 10 or Windows 11 to use the commands below.</span></div>
<div id="wb_Text5" style="position:absolute;left:409px;top:1376px;width:707px;height:32px;z-index:8;">
<span style="color:#000000;font-family:'Times New Roman';font-size:29px;"><strong>Command</strong></span></div>
<div id="wb_Text6" style="position:absolute;left:406px;top:1423px;width:859px;height:35px;text-align:justify;z-index:9;">
<span style="color:#000000;font-family:'Times New Roman';font-size:16px;">Open PowerShell or Windows Command Prompt in administrator mode by right-clicking and selecting &quot;Run as administrator&quot;, enter the wsl --install command, then restart your machine.</span></div>
<div id="wb_Text8" style="position:absolute;left:409px;top:1527px;width:859px;height:37px;text-align:justify;z-index:10;">
<span style="color:#000000;font-family:'Times New Roman';font-size:17px;"><strong>For more informatio see</strong></span><span style="color:#000000;font-family:'Times New Roman';font-size:16px;">: <a href="https://learn.microsoft.com/en-us/windows/wsl/install">https://learn.microsoft.com/en-us/windows/wsl/install<br></a></span></div>
<div id="wb_Text7" style="position:absolute;left:406px;top:1561px;width:859px;height:96px;text-align:justify;z-index:11;">
<ul style="list-style-type:disc;">
<li style="margin:0 0 0 24px;">You can after installing wsl close your CMD and open it again and type type "wsl" and hit "Enter". 
</li>
</ul>
<br>
<ul style="list-style-type:disc;">
<li style="margin:0 0 0 24px;">WSL stores your Windows drives in the /mnt folder, with the name of the drive as a subfolder. For example your C:\ drive will be present at /mnt/c/ for you to use. Keeping this in mind, you can swap to your specific folder like so:cd /mnt/e/username/folder1/folder2
</li>
</ul></div>
<div id="wb_Fastqc_(WSL_Linux_ubuntu)" style="position:absolute;left:370px;top:1677px;width:36px;height:33px;z-index:12;">
<div id="Fastqc_(WSL_Linux_ubuntu)" style="visibility:hidden;"></div>
</div>
<div id="wb_Text10" style="position:absolute;left:409px;top:1730px;width:859px;height:71px;text-align:justify;z-index:13;">
<span style="color:#000000;font-family:Arial;font-size:16px;">Linux, with its robust capabilities, is a preferred choice for many in the scientific community. Installing FastQC on a Linux system is straightforward. Follow these steps to ensure a smooth installation: Update Your System: Start by updating your systemâ€™s package list to have the latest software references. Open your terminal and enter:<br></span></div>
<div id="wb_Breadcrumb1" style="position:absolute;left:403px;top:13px;width:862px;height:39px;z-index:14;vertical-align:top;">
<ul id="Breadcrumb1">
<li><a href=""><i class="fa fa-home"></i>   Home</a></li>
<li><a href="">Metagenomics </a></li>
</ul>
</div>
<div id="wb_Breadcrumb2" style="position:absolute;left:1092px;top:13px;width:173px;height:39px;z-index:15;vertical-align:top;">
<ul id="Breadcrumb2">
<li><a href="https://github.com/javad30" target="_blank"><i class="fa fa-brands fa-github"></i>See on GITHUB</a></li>
</ul>
</div>
<div id="wb_Text11" style="position:absolute;left:409px;top:1855px;width:859px;height:36px;text-align:justify;z-index:16;">
<span style="color:#000000;font-family:Arial;font-size:16px;">Install FastQC: With your package list updated, proceed to install FastQC. In the terminal, type:<br></span></div>
<div id="wb_Text12" style="position:absolute;left:409px;top:1946px;width:859px;height:36px;text-align:justify;z-index:17;">
<span style="color:#000000;font-family:Arial;font-size:16px;">This command initiates the download and installation of FastQC.<br></span></div>
<div id="wb_Text14" style="position:absolute;left:409px;top:1982px;width:859px;height:37px;text-align:justify;z-index:18;">
<span style="color:#000000;font-family:Arial;font-size:17px;"><strong>Verification</strong></span><span style="color:#000000;font-family:Arial;font-size:16px;">: To confirm that FastQC has been installed correctly, run the following in your terminal:<br></span></div>
<div id="wb_Text15" style="position:absolute;left:409px;top:2065px;width:859px;height:71px;text-align:justify;z-index:19;">
<span style="color:#000000;font-family:Arial;font-size:16px;">This should display the help information and version number, indicating a successful installation.By completing these steps, FastQC will be installed and ready for use on your Linux system, setting the stage for high-quality sequence data analysis.<br></span></div>
<div id="wb_Text17" style="position:absolute;left:405px;top:1470px;width:837px;height:18px;z-index:20;">
<p>wsl --install</p></div>
<div id="wb_Text18" style="position:absolute;left:408px;top:1801px;width:837px;height:18px;z-index:21;">
<p>sudo apt update</p></div>
<div id="wb_Text19" style="position:absolute;left:408px;top:1891px;width:837px;height:18px;z-index:22;">
<p>sudo apt -y install fastqc</p></div>
<div id="wb_Text20" style="position:absolute;left:408px;top:2019px;width:837px;height:18px;z-index:23;">
<p>fastqc--help</p></div>
<div id="wb_Icon1" style="position:absolute;left:12px;top:19px;width:33px;height:25px;text-align:center;z-index:24;">
<a href="#Installation"><div id="Icon1"><i class="fa fa-house-chimney"></i></div></a></div>
<div id="wb_Text13" style="position:absolute;left:407px;top:62px;width:707px;height:32px;z-index:25;">
<span style="color:#000000;font-family:'Times New Roman';font-size:29px;"><strong>Home</strong></span></div>
<div id="wb_Home" style="position:absolute;left:370px;top:53px;width:52px;height:51px;z-index:26;">
<div id="Home" style="visibility:hidden;"></div>
</div>
<div id="wb_Text16" style="position:absolute;left:409px;top:104px;width:859px;height:455px;text-align:justify;z-index:27;">
<p>This genome-centric metagenomics workshop will teach you how to obtain provisional whole genomes of individual populations from a mixed microbial community using metagenomics. The workshop has three parts:</p>
<br>
<br>
<p><span style="font-weight:bold;">Part I</span>: Participants will learn to install required software, check raw sequence read quality, perform read quality control, and trim their sequence data. Participants will also learn how to upload and dowload sequence files from online archive databases. </p>
<br>
<br>
<br>
<p><span style="font-weight:bold;">Part II</span>: In this part you learn how to assemble and annotate contigs, bin contigs into provisional whole genome sequences.</p>
<br>
<br>
<p><span style="font-weight:bold;">Part III</span>:&nbsp; Participants will learn to how to extract taxonomic information, functional annotations, and pathway information for each binned genomes.</p>
<br>
<p style="font-size:20px;line-height:23px;font-weight:bold;">Workshop Topic Highlights</p>
<p style="font-size:20px;line-height:23px;font-weight:bold;">&nbsp;</p>
<ul style="list-style-type:disc;">
<li style="margin:0 0 0 24px;">Metagenome reads quality check and quality control
</li>
<li style="margin:0 0 0 24px;">Assemble quality controlled reads into contigs
</li>
<li style="margin:0 0 0 24px;">Annotate assembled contigs
</li>
<li style="margin:0 0 0 24px;">Map quality controlled reads onto contigs
</li>
<li style="margin:0 0 0 24px;">Bin contigs into provisional genome bins
</li>
<li style="margin:0 0 0 24px;">Extract taxonomic, functional, and pathway annotations for each binned genomes
</li>
</ul>
<br></div>
<div id="wb_Image1" style="position:absolute;left:411px;top:580px;width:737px;height:371px;z-index:28;">
<img src="images/1.png" id="Image1" alt="" width="737" height="370"></div>
<div id="wb_Text21" style="position:absolute;left:403px;top:988px;width:707px;height:32px;z-index:29;">
<span style="color:#000000;font-family:'Times New Roman';font-size:29px;"><strong>Software installation</strong></span></div>
<div id="wb_Text22" style="position:absolute;left:408px;top:1016px;width:859px;height:77px;z-index:30;">
<span style="color:#000000;font-family:'Times New Roman';font-size:16px;"><br></span><span style="color:#000000;font-family:'Times New Roman';font-size:19px;">In this section you would learn how to install several metagenomics software including used for Metagenomics assembly, binning, gene prediction and annotation and strain-level analysis.<br></span></div>
<img src="images/img0003.png" id="Banner1" alt=" Install WSL" style="border-width:0;position:absolute;left:403px;top:1109px;width:856px;height:44px;z-index:31;" width="856" height="44">
<img src="images/img0004.png" id="Banner2" alt="Fastqc (WSL, Linux ubuntu)" style="border-width:0;position:absolute;left:403px;top:1666px;width:856px;height:44px;z-index:32;" width="856" height="44">
<div id="wb_Text1" style="position:absolute;left:409px;top:2434px;width:859px;height:269px;text-align:justify;z-index:33;">
<span style="color:#000000;font-family:'Times New Roman';font-size:19px;">First install brew by either going to <strong>https://brew.sh/ </strong>and follow the instruction on how to install brew on your system. <br><br>Or copy and past the following command in your terminal:&nbsp;&nbsp; <br><br>/bin/bash -c &quot;$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)&quot;<br><br><br>Once you install brew then run the following code to install fastqc (will take up to 5 mins to install fastqc):<br><br>brew install fastqc<br><br>After installing fastqc test your fastqc by running: fastqc --help</span></div>
<img src="images/img0005.png" id="Banner3" alt="Fastqc (Mac)" style="border-width:0;position:absolute;left:407px;top:2366px;width:856px;height:44px;z-index:34;" width="856" height="44">
<div id="wb_Image2" style="position:absolute;left:573px;top:2368px;width:41px;height:41px;z-index:35;">
<img src="images/4.png" id="Image2" alt="" width="41" height="41"></div>
<div id="wb_Fastqc__Mac_" style="position:absolute;left:370px;top:2374px;width:36px;height:33px;z-index:36;">
<div id="Fastqc__Mac_" style="visibility:hidden;"></div>
</div>
<div id="wb_Image4" style="position:absolute;left:564px;top:1110px;width:76px;height:42px;z-index:37;">
<img src="images/2.jpg" id="Image4" alt="" width="76" height="43"></div>
<div id="wb_DropdownMenu1" style="position:absolute;left:0px;top:70px;width:233px;height:648px;z-index:38;" data-wb-spy="affix" >
<div id="DropdownMenu1" class="DropdownMenu1" style="width:100%;height:auto !important;">
<div class="container">
<div class="navbar-header">
<button title="Dropdown Menu" type="button" class="navbar-toggle" data-bs-toggle="collapse" data-bs-target=".DropdownMenu1-navbar-collapse">
<span class="icon-bar"></span>
<span class="icon-bar"></span>
<span class="icon-bar"></span>
</button>
</div>
<div class="DropdownMenu1-navbar-collapse collapse">
<ul class="nav navbar-nav">
<li class="nav-item">
<a href="#Home" class="nav-link">Home</a>
</li>
<li class="nav-item dropdown">
<a href="#" class="dropdown-toggle" data-bs-placement="bottom-start" data-bs-toggle="dropdown">Installing software<span class="arrow"></span></a>
<ul class="dropdown-menu">
<li class="nav-item dropdown-item">
<a href="#Install_WSL" class="nav-link">Install WSL</a>
</li>
<li class="nav-item dropdown-item">
<a href="#Fastqc_(WSL_Linux_ubuntu)" class="nav-link">Fastqc (WSL, Linux)</a>
</li>
<li class="nav-item dropdown-item">
<a href="#Fastqc__Mac_" class="nav-link">Fastqc (Mac)</a>
</li>
<li class="nav-item dropdown-item">
<a href="#bowtie2" class="nav-link">bowtie2</a>
</li>
</ul>
</li>
<li class="nav-item dropdown">
<a href="#" class="dropdown-toggle" data-bs-placement="bottom-start" data-bs-toggle="dropdown">Workshop day 1<span class="arrow"></span></a>
<ul class="dropdown-menu">
<li class="nav-item dropdown-item">
<a href="#Obtaining_sequence_data" class="nav-link">Obtaining sequence data</a>
</li>
<li class="nav-item dropdown-item">
<a href="#Trimming_sequence_data" class="nav-link">Trimming sequence data</a>
</li>
<li class="nav-item dropdown-item">
<a href="#Removing_host_contamination" class="nav-link">Removing host contamination</a>
</li>
</ul>
</li>
</ul>
</div>
</div>
</div></div>
<div id="wb_Image5" style="position:absolute;left:589px;top:2125px;width:406px;height:210px;z-index:39;">
<img src="images/3.png" id="Image5" alt="" width="406" height="211"></div>
<img src="images/img0006.png" id="Banner4" alt="Obtaining sequence data" style="border-width:0;position:absolute;left:405px;top:4133px;width:856px;height:44px;z-index:40;" width="856" height="44">
<div id="wb_Obtaining_sequence_data" style="position:absolute;left:370px;top:4144px;width:36px;height:33px;z-index:41;">
<div id="Obtaining_sequence_data" style="visibility:hidden;"></div>
</div>
<div id="wb_Text9" style="position:absolute;left:406px;top:4577px;width:859px;height:620px;text-align:justify;z-index:42;">
<p><span style="font-weight:bold;">SRA: </span>You can either bring your sequence of intrest (above) or dowload the sequence from online resources. One of the most useful online resources to download archive sequence files is using NCBI SRA. To do so:</p>
<br>
<ul style="list-style-type:upper-latin;">
<li style="margin:0 0 0 24px;">Go to NCBI SRA website (https://www.ncbi.nlm.nih.gov/sra) and search for species/project. <br/><br/>
</li>
<li style="margin:0 0 0 24px;">After finding the Biom Project/sequence that you want compy the SRA ID under RUN tab.<br/><br/>
</li>
<li style="margin:0 0 0 24px;">Go to your termina (all users (Mac, Ubunta) and install SRA downloader using below command. <br/><br/><br/><br/>
</li>
<li style="margin:0 0 0 24px;">After installing sra-toolkit use below code to dowaload the sequence with your SRA id that you found before (step C). For example I was interested in Human microbiome and found this sequnce on here (https://www.ncbi.nlm.nih.gov/sra/SRX25254909[accn]) and the SRA id was <span style="font-weight:bold;">SRR29754043. </span>	<br/><br/><br/>&nbsp;
</li>
<li style="margin:0 0 0 24px;">Now we have to change the SRA file to fastq file. Go to the downloaded folder using cd command then use below code to change sra to fastq file.<br/><br/><br/><br/><br/><br/><br/>
</li>
<li style="margin:0 0 0 24px;">Now lest check the fastq file quality using fastqc software.
</li>
</ul>
<br>
<br>
<br>
<br></div>
<div id="wb_Text23" style="position:absolute;left:409px;top:4747px;width:837px;height:18px;z-index:43;">
<p>sudo apt install sra-toolkit</p></div>
<div id="wb_Text24" style="position:absolute;left:411px;top:4876px;width:837px;height:18px;z-index:44;">
<p>prefetch SRR29754043</p></div>
<div id="wb_Text25" style="position:absolute;left:408px;top:5029px;width:837px;height:18px;z-index:45;">
<p>fastq-dump --split-files SRR29754043&nbsp;&nbsp; -O your/output/file</p></div>
<div id="wb_Text26" style="position:absolute;left:407px;top:4986px;width:837px;height:18px;z-index:46;">
<p>cd SRR29754043</p></div>
<div id="wb_Text27" style="position:absolute;left:408px;top:5127px;width:837px;height:54px;z-index:47;">
<p>cd your/output/folder</p>
<br>
<p>fastqc &nbsp;*fastq&nbsp;&nbsp; -O SRA_fastq</p></div>
<img src="images/img0007.png" id="Banner5" alt="Trimming sequence data" style="border-width:0;position:absolute;left:407px;top:5243px;width:856px;height:44px;z-index:48;" width="856" height="44">
<div id="wb_Trimming_sequence_data" style="position:absolute;left:367px;top:5243px;width:36px;height:33px;z-index:49;">
<div id="Trimming_sequence_data" style="visibility:hidden;"></div>
</div>
<div id="wb_Text28" style="position:absolute;left:409px;top:5303px;width:859px;height:744px;z-index:50;">
<p>There are a variety of software for Trimming (e.g, removing adapter). Adapter sequences should be removed from reads because they interfere with downstream analyses, such as alignment of reads to a reference. SRA include a few changes to the sequnce files, that are not compatible with some of the analysis. However, for triming you can use your SRA sequences for practice.</p>
<br>
<ul style="list-style-type:upper-latin;">
<li style="margin:0 0 0 24px;">Find your adaptor (fastq or terminal). In terminal use grep command and input the sequence file that you think might be the adapter to see if you can find it. for example, using fastqc I found that my sequence from SRA has NexteraPE-PE adaptor and I used grep to see if this is true. <br/><br/><br/><br/><br/><br/><br/><br/><br/>
</li>
<li style="margin:0 0 0 24px;">Go http://www.usadellab.org/cms/?page=trimmomatic. <br/><br/>
</li>
<li style="margin:0 0 0 24px;">Download Trimmomatic (select Version 0.39: binary)<br/><br/>
</li>
<li style="margin:0 0 0 24px;">Open the downloaded folder and go to <span style="font-weight:bold;font-style:italic;">adapters </span>folder and open <span style="font-weight:bold;font-style:italic;">NexteraPE-PE.fa </span>file (or any other adaptor that you have based on step A). Copy the sequnces.
</li>
<li style="margin:0 0 0 24px;">Then run the below code in your terminal 
</li>
</ul>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<p>It works with FASTQ (using phred + 33 or phred + 64 quality scores, depending on the Illumina pipeline used), either uncompressed or gzipp'ed FASTQ. Use of gzip format is determined based on the .gz extension.</p>
<br>
<p>For single-ended data, one input and one output file are specified, plus the processing steps. For paired-end data, two input files are specified, and 4 output files, 2 for the 'paired' output where both reads survived the processing, and 2 for corresponding 'unpaired' output where a read survived, but the partner read did not.</p></div>
<div id="wb_Text29" style="position:absolute;left:400px;top:5475px;width:837px;height:18px;z-index:51;">
<p>grep AGATGTGTATAAGAGACAG&nbsp; SRR29754043_2.fastq</p></div>
<div id="wb_Text30" style="position:absolute;left:400px;top:5812px;width:837px;height:54px;z-index:52;">
<p>java -jar trimmomatic-0.39.jar PE S28_R1.gz S28_R2.gz output_forward_paired.fq.gz output_forward_unpaired.fq.gz output_reverse_paired.fq.gz output_reverse_unpaired.fq.gz ILLUMINACLIP:NexteraPE-PE.fa:2:30:10:2:True LEADING:3 TRAILING:3 MINLEN:36</p></div>
<img src="images/img0008.png" id="Banner6" alt="Removing host contamination" style="border-width:0;position:absolute;left:396px;top:6155px;width:856px;height:44px;z-index:53;" width="856" height="44">
<div id="wb_Removing_host_contamination" style="position:absolute;left:364px;top:6155px;width:36px;height:33px;z-index:54;">
<div id="Removing_host_contamination" style="visibility:hidden;"></div>
</div>
<img src="images/img0009.png" id="Banner7" alt="Bowtie2" style="border-width:0;position:absolute;left:403px;top:2734px;width:856px;height:44px;z-index:55;" width="856" height="44">
<div id="wb_bowtie2" style="position:absolute;left:367px;top:2734px;width:36px;height:33px;z-index:56;">
<div id="bowtie2" style="visibility:hidden;"></div>
</div>
<div id="wb_Text31" style="position:absolute;left:409px;top:2767px;width:859px;height:496px;z-index:57;">
<span style="color:#000000;font-family:'Times New Roman';font-size:19px;"><br>Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences. It is particularly good at aligning reads of about 50 up to 100s of characters to relatively long (e.g. mammalian) genomes. In my opinion Bowtie2 is probably the most improtant software in genomics (Cited 50K)! Read more about Bowtie2 here: https://bowtie-bio.sourceforge.net/bowtie2/manual.shtml.<strong><br><br></strong>There are multiple ways to download and isntall Bowtie2. The two most common ways: <strong><br><br><br>a) Install bowtie2 using conda<br><br><br><br><br>b) Install bowtie2 using a package manager (recommended!)<br><br></strong><br><br><br><br><br><strong>C) Check installation</strong><br><br></span></div>
<div id="wb_Text33" style="position:absolute;left:409px;top:3001px;width:837px;height:18px;z-index:58;">
<p>conda install -c bioconda bowtie2</p></div>
<div id="wb_Text32" style="position:absolute;left:405px;top:3108px;width:837px;height:54px;z-index:59;">
<p>sudo apt update</p>
<br>
<p>sudo apt install bowtie2</p></div>
<div id="wb_Text34" style="position:absolute;left:405px;top:3325px;width:837px;height:18px;z-index:60;">
<p>bowtie2 --help</p></div>
<div id="wb_Text36" style="position:absolute;left:398px;top:6219px;width:862px;height:1880px;text-align:justify;z-index:61;">
<p>Sometimes we have host DNA and we need to remove the host DNA before doing any downstream analysis. For example, if I am looking at mice gut microbiome, I do not need mice DNA, however, during the DNA extractions some of the mice gut intestinal epithelial DNA will be sequenced as well. To remove that we are going to use Bowtie2. Bowtie2 is a reference base alinement and has huge applications in RNA and WG sequencing. To remove host DNA run below code. We will follow the instructions provided here:</p>
<p style="font-size:20px;line-height:23px;font-weight:bold;">https://www.metagenomics.wiki/tools/short-read/remove-host-sequences</p>
<p style="font-size:20px;line-height:23px;font-weight:bold;">&nbsp;</p>
<ul style="font-size:20px;line-height:23px;list-style-type:upper-latin;">
<li style="margin:0 0 0 24px;"><span style="font-weight:normal;">To delete host genome first we need to download host DNA genome (e.g,. from NCBI (below pic)).</span><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/><br/>
</li>
<li style="margin:0 0 0 24px;"><span style="font-weight:normal;">After downloading the host genome (here we use tomato because the samples are tomato microbiome) we need to index the host genome using below code. </span><br/><br/><br/><br/><br/>
</li>
<li style="margin:0 0 0 24px;"><span style="font-weight:normal;">The above code will generate several files. Then we use bowtie to remove host genome. </span>
</li>
</ul>
<p style="font-size:20px;line-height:23px;font-weight:bold;">&nbsp;</p>
<p style="font-size:20px;line-height:23px;font-weight:bold;">&nbsp;</p>
<p style="font-size:20px;line-height:23px;font-weight:bold;">&nbsp;</p>
<p style="font-size:20px;line-height:23px;font-weight:bold;">&nbsp;</p>
<p style="font-size:20px;line-height:23px;font-weight:bold;">&nbsp;</p>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<p style="color:#8B0000;"><span style="font-size:20px;line-height:23px;font-weight:bold;">This will take hours to finish (for </span><span style="font-weight:bold;color:#800000;">Solanaceae</span><span style="color:#000000;">&nbsp;</span><span style="background-color:#FFFFFF;font-size:20px;line-height:23px;font-weight:bold;">g</span><span style="font-size:20px;line-height:23px;font-weight:bold;">enome) ! --threads change this based on your computer cores. </span><span style="color:#000000;">The command should print many lines of output then quit. The current directory will contain four new files that all start with </span><span style="font-family:Arial;font-size:16px;line-height:18px;color:#000000;">Solanaceae</span><span style="font-family:Arial;font-size:16px;line-height:18px;color:#000000;">&nbsp;</span><span style="color:#000000;">and end with .1.bt2, .2.bt2, .3.bt2, .4.bt2, .rev.1.bt2, and .rev.2.bt2. These files constitute the index - you're done!</span></p>
<br>
<p>Useful links:</p>
<p>Building reference (e.g, Multilple E.coli): https://www.metagenomics.wiki/tools/bowtie2/index</p>
<p>https://open.bioqueue.org/home/knowledge/showKnowledge/sig/bowtie2</p>
<p>Archives indexes: https://bowtie-bio.sourceforge.net/bowtie2/news.shtml</p>
<br>
<br>
<br>
<br>
<br>
<br>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;color:#800000;">Now let's change the file names for downstream analyses.</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;color:#800000;">Now lets change SAM file to BAM file. </p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;color:#800000;">If you do not have samtools install this first</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<br>
<br>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;color:#800000;">Change Bam to sorted bam</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;">&nbsp;</p>
<p style="font-weight:bold;color:#800000;">Creating a BAM index file</p>
<br>
<br>
<br>
<br>
<br>
<br>
<br></div>
<div id="wb_Text37" style="position:absolute;left:406px;top:6795px;width:731px;height:36px;z-index:62;">
<p>bowtie2&nbsp; -x 0_Host_genome/Solanaceae/Solanaceae&nbsp; -1 S28_R1.gz&nbsp; -2 S28_R2.gz&nbsp; --un-conc-gz&nbsp;&nbsp; S28_removed_Solanaceae&nbsp; -S Mapped_and_unmapped_Solanaceae.sam&nbsp; -p 20 --very-sensitive-local</p></div>
<div id="wb_Text35" style="position:absolute;left:403px;top:6694px;width:837px;height:18px;z-index:63;">
<p> bowtie2-build Solanaceae.fna&nbsp; Solanaceae --threads 20</p></div>
<div id="wb_Image7" style="position:absolute;left:564px;top:6413px;width:457px;height:174px;z-index:64;">
<img src="images/6.png" id="Image7" alt="" width="457" height="227"></div>
<div id="wb_Image11" style="position:absolute;left:484px;top:6875px;width:643px;height:215px;z-index:65;">
<img src="images/7.png" id="Image11" alt="" width="643" height="246"></div>
<div id="wb_Image6" style="position:absolute;left:500px;top:5526px;width:652px;height:79px;z-index:66;">
<img src="images/5.png" id="Image6" alt="" width="652" height="79"></div>
<div id="wb_Image8" style="position:absolute;left:451px;top:7344px;width:767px;height:87px;z-index:67;">
<img src="images/10.png" id="Image8" alt="" width="767" height="87"></div>
<div id="wb_Text38" style="position:absolute;left:405px;top:7541px;width:837px;height:36px;z-index:68;">
<p>mv S28_removed_Solanaceae.1&nbsp; S28_host_removed_R1.fastq.gz</p>
<p>mv S28_removed_Solanaceae.2&nbsp;&nbsp; S28_host_removed_R2.fastq.gz</p></div>
<div id="wb_Text57" style="position:absolute;left:403px;top:7703px;width:837px;height:18px;z-index:69;">
<p>conda install bioconda::samtools</p></div>
<div id="wb_Text41" style="position:absolute;left:398px;top:7757px;width:837px;height:18px;z-index:70;">
<p> samtools view -bS Mapped_and_unmapped_Solanaceae.sam&nbsp; &gt; S28_mapped_and_unmapped.bam</p></div>
<div id="wb_Text58" style="position:absolute;left:400px;top:8004px;width:837px;height:18px;z-index:71;">
<p>samtools index S28_sorted.bam S28_sorted.bai</p></div>
<div id="wb_Text59" style="position:absolute;left:400px;top:7875px;width:837px;height:18px;z-index:72;">
<p>samtools sort S28_mapped_and_unmapped.bam -o S28_sorted.bam</p></div>
<div id="wb_Text45" style="position:absolute;left:1701px;top:4190px;width:357px;height:15px;z-index:73;">
&nbsp;</div>
<div id="wb_Text39" style="position:absolute;left:406px;top:4205px;width:859px;height:248px;z-index:74;">
<span style="color:#000000;font-family:'Times New Roman';font-size:19px;">Test data: We will pratice with two types of datasets. A. Online and B. Stored data.<br><br><br>To download my own data:<br><br><br><strong>Dropbox:<br><br></strong>https://www.dropbox.com/scl/fo/syejnazitw7lzkj5rcot2/AKIDndJnXSt19QbwLuhAIfs?rlkey=ykh9v933296fhspndkezanocq&amp;st=ho4vjazf&amp;dl=0<br><br></span></div>

</body>
</html>