// search index for WYSIWYG Web Builder
var database_length = 0;

function SearchPage(url, title, keywords, description)
{
   this.url = url;
   this.title = title;
   this.keywords = keywords;
   this.description = description;
   return this;
}

function SearchDatabase()
{
   database_length = 0;
   this[database_length++] = new SearchPage("index.html", "Untitled Page", " What is Windows Subsystem for Linux  WSL ?  The Windows Subsystem for Linux  WSL  is a feature of the Windows operating system that enables you to run a Linux file system, along with Linux command-line tools and GUI apps, directly on Windows, alongside your traditional Windows desktop and apps   Prerequisites  You must be running Windows 10 or Windows 11 to use the commands below  Command  Open PowerShell or Windows Command Prompt in administrator mode by right-clicking and selecting \"Run as administrator\", enter the wsl --install command, then restart your machine  For more informatio see  https //learn.microsoft.com/en-us/windows/wsl/install   You can after installing wsl close your CMD and open it again and type type \"wsl\" and hit \"Enter\"   WSL stores your Windows drives in the /mnt folder, with the name of the drive as a subfolder For example your C   drive will be present at /mnt/c/ for you to use Keeping this in mind, you can swap to your specific folder like so cd /mnt/e/username/folder1/folder2  Linux, with its robust capabilities, is a preferred choice for many in the scientific community Installing FastQC on a Linux system is straightforward Follow these steps to ensure a smooth installation  Update Your System  Start by updating your systemâ€™s package list to have the latest software references Open your terminal and enter    Install FastQC  With your package list updated, proceed to install FastQC In the terminal, type    This command initiates the download and installation of FastQC   Verification  To confirm that FastQC has been installed correctly, run the following in your terminal    This should display the help information and version number, indicating a successful installation.By completing these steps, FastQC will be installed and ready for use on your Linux system, setting the stage for high-quality sequence data analysis   wsl --install  sudo apt update  sudo apt -y install fastqc  fastqc--help  Home  This genome-centric metagenomics workshop will teach you how to obtain provisional whole genomes of individual populations from a mixed microbial community using metagenomics The workshop has three parts    Part I  Participants will learn to install required software, check raw sequence read quality, perform read quality control, and trim their sequence data Participants will also learn how to upload and dowload sequence files from online archive databases     Part II  In this part you learn how to assemble and annotate contigs, bin contigs into provisional whole genome sequences   Part III   Participants will learn to how to extract taxonomic information, functional annotations, and pathway information for each binned genomes  Workshop Topic Highlights  Metagenome reads quality check and quality control Assemble quality controlled reads into contigs Annotate assembled contigs Map quality controlled reads onto contigs Bin contigs into provisional genome bins Extract taxonomic, functional, and pathway annotations for each binned genomes   Software installation   In this section you would learn how to install several metagenomics software including used for Metagenomics assembly, binning, gene prediction and annotation and strain-level analysis   First install brew by either going to https //brew.sh/ and follow the instruction on how to install brew on your system   Or copy and past the following command in your terminal      /bin/bash -c \"$ curl -fsSL https //raw.githubusercontent.com/Homebrew/install/HEAD/install.sh \"   Once you install brew then run the following code to install fastqc  will take up to 5 mins to install fastqc    brew install fastqc  After installing fastqc test your fastqc by running  fastqc --help  SRA  You can either bring your sequence of intrest  above  or dowload the sequence from online resources One of the most useful online resources to download archive sequence files is using NCBI SRA To do so   Go to NCBI SRA website  https //www.ncbi.nlm.nih.gov/sra  and search for species/project  After finding the Biom Project/sequence that you want compy the SRA ID under RUN tab. Go to your termina  all users  Mac, Ubunta  and install SRA downloader using below command  After installing sra-toolkit use below code to dowaload the sequence with your SRA id that you found before  step C  For example I was interested in Human microbiome and found this sequnce on here  https //www.ncbi.nlm.nih.gov/sra/SRX25254909 accn   and the SRA id was SRR29754043    Now we have to change the SRA file to fastq file Go to the downloaded folder using cd command then use below code to change sra to fastq file. Now lest check the fastq file quality using fastqc software      sudo apt install sra-toolkit  prefetch SRR29754043  fastq-dump --split-files SRR29754043   -O your/output/file  cd SRR29754043  cd your/output/folder  fastqc   fastq   -O SRA_fastq  There are a variety of software for Trimming  e.g, removing adapter  Adapter sequences should be removed from reads because they interfere with downstream analyses, such as alignment of reads to a reference SRA include a few changes to the sequnce files, that are not compatible with some of the analysis However, for triming you can use your SRA sequences for practice  Find your adaptor  fastq or terminal  In terminal use grep command and input the sequence file that you think might be the adapter to see if you can find it for example, using fastqc I found that my sequence from SRA has NexteraPE-PE adaptor and I used grep to see if this is true  Go http //www.usadellab.org/cms/?page=trimmomatic  Download Trimmomatic  select Version 0.39  binary  Open the downloaded folder and go to adapters folder and open NexteraPE-PE.fa file  or any other adaptor that you have based on step A  Copy the sequnces Then run the below code in your terminal         It works with FASTQ  using phred + 33 or phred + 64 quality scores, depending on the Illumina pipeline used , either uncompressed or gzipp'ed FASTQ Use of gzip format is determined based on the .gz extension  For single-ended data, one input and one output file are specified, plus the processing steps For paired-end data, two input files are specified, and 4 output files, 2 for the 'paired' output where both reads survived the processing, and 2 for corresponding 'unpaired' output where a read survived, but the partner read did not  grep AGATGTGTATAAGAGACAG  SRR29754043_2.fastq  java -jar trimmomatic-0.39.jar PE S28_R1.gz S28_R2.gz output_forward_paired.fq.gz output_forward_unpaired.fq.gz output_reverse_paired.fq.gz output_reverse_unpaired.fq.gz ILLUMINACLIP NexteraPE-PE.fa 2 30 10 2 True LEADING 3 TRAILING 3 MINLEN 36   Bowtie 2 is an ultrafast and memory-efficient tool for aligning sequencing reads to long reference sequences It is particularly good at aligning reads of about 50 up to 100s of characters to relatively long  e.g mammalian  genomes In my opinion Bowtie2 is probably the most improtant software in genomics  Cited 50K ! Read more about Bowtie2 here  https //bowtie-bio.sourceforge.net/bowtie2/manual.shtml  There are multiple ways to download and isntall Bowtie2 The two most common ways     a  Install bowtie2 using conda     b  Install bowtie2 using a package manager  recommended!        C  Check installation    conda install -c bioconda bowtie2  sudo apt update  sudo apt install bowtie2  bowtie2 --help  Sometimes we have host DNA and we need to remove the host DNA before doing any downstream analysis For example, if I am looking at mice gut microbiome, I do not need mice DNA, however, during the DNA extractions some of the mice gut intestinal epithelial DNA will be sequenced as well To remove that we are going to use Bowtie2 Bowtie2 is a reference base alinement and has huge applications in RNA and WG sequencing To remove host DNA run below code We will follow the instructions provided here  https //www.metagenomics.wiki/tools/short-read/remove-host-sequences  To delete host genome first we need to download host DNA genome  e.g, from NCBI  below pic  . After downloading the host genome  here we use tomato because the samples are tomato microbiome  we need to index the host genome using below code  The above code will generate several files Then we use bowtie to remove host genome                   This will take hours to finish  for Solanaceae genome  ! --threads change this based on your computer cores The command should print many lines of output then quit The current directory will contain four new files that all start with Solanaceae and end with .1.bt2, .2.bt2, .3.bt2, .4.bt2, .rev.1.bt2, and .rev.2.bt2 These files constitute the index - you're done!  Useful links  Building reference  e.g, Multilple E.coli   https //www.metagenomics.wiki/tools/bowtie2/index https //open.bioqueue.org/home/knowledge/showKnowledge/sig/bowtie2 Archives indexes  https //bowtie-bio.sourceforge.net/bowtie2/news.shtml         Now let's change the file names for downstream analyses       Now lets change SAM file to BAM file   If you do not have samtools install this first       Change Bam to sorted bam      Creating a BAM index file         bowtie2  -x 0_Host_genome/Solanaceae/Solanaceae  -1 S28_R1.gz  -2 S28_R2.gz  --un-conc-gz   S28_removed_Solanaceae  -S Mapped_and_unmapped_Solanaceae.sam  -p 20 --very-sensitive-local   bowtie2-build Solanaceae.fna  Solanaceae --threads 20  mv S28_removed_Solanaceae.1  S28_host_removed_R1.fastq.gz mv S28_removed_Solanaceae.2   S28_host_removed_R2.fastq.gz  conda install bioconda  samtools   samtools view -bS Mapped_and_unmapped_Solanaceae.sam    S28_mapped_and_unmapped.bam  samtools index S28_sorted.bam S28_sorted.bai  samtools sort S28_mapped_and_unmapped.bam -o S28_sorted.bam    Test data  We will pratice with two types of datasets A Online and B Stored data   To download my own data    Dropbox   https //www.dropbox.com/scl/fo/syejnazitw7lzkj5rcot2/AKIDndJnXSt19QbwLuhAIfs?rlkey=ykh9v933296fhspndkezanocq&st=ho4vjazf&dl=0     ", "");
   return this;
}
